frontend/src/hooks/useEventListener.ts

```diff
@@ -34,5 +45,5 @@ export function useEventListener<T extends Target, K extends keyof EventMap<T>>(
     return () => {
       target.removeEventListener(type, eventListener, options);
     };
-  }, [type, target, options]);
+  }, [type, targetValue, options]);
```

Comment:
If we are passing the RefObject, wont this not trigger re-renders when it changes?

Although if this fixes the underlying issues, I trust this is the right change.

Reply:
mutating the ref.current value won't trigger re-renders. But the ref object itself does. I'm basing this because you would normally get a warning when using ref.current in useEffect

`Mutable values like 'ref.current' aren't valid dependencies because mutating them doesn't re-render the component. (react-hooks/exhaustive-deps)`

But I've searched online and see that a common solution is to use `useCallback`, but I think that would require a lot of code modification. Do you think I should look more in this direction?

Reply:
if this fixes it, i am totally ok with it! i think this was a bug that was already there, but I added ReactCompiler which memoizes a lot such that subtle bugs like this could crop up.

also, you might have fixed this https://github.com/marimo-team/marimo/issues/2940 (thank you!)